WordPress/gutenberg
# [Inserter]: Adjust order of theme blocks and reorder inserter items

Resolves: https://github.com/WordPress/gutenberg/issues/36547

This PR prioritizes some blocks by just changing their registration order (existing behavior).
In addition I made a change to show its block's variations right after their source block. Previously all source blocks were shown and all their variations followed, which doesn't make much sense to me :) 

## Testing instructions
1. In a page open the main inserter and go to `theme` block category
2. Observe that the block variations are after their source block and also that the blocks we changed registration order are at the top of the list.
